### PR TITLE
[runtime] Comment two special cases that triggers static analysis tools

### PR DESCRIPTION
**Pull request by spouliot**

### CODE REVIEW

File: tools/mtouch/simlauncher.mm [hunk 1]
```diff
@@ -40,6 +40,14 @@ void xamarin_setup_impl ()
 int
 main (int argc, char** argv)
 {
+	// There's a potentially uncaugth exception here, it
```
Comment (Therzok) [inline tools/mtouch/simlauncher.mm]:
```suggestion
	// There's a potentially uncaught exception here, it
```

Comment (Therzok) [reply tools/mtouch/simlauncher.mm]: Typo and possible wording improvement? :D

File: tools/mtouch/simlauncher.mm [hunk 2]
```diff
@@ -40,6 +40,14 @@ void xamarin_setup_impl ()
 int
 main (int argc, char** argv)
 {
+	// There's a potentially uncaught exception here, which:
+	// 1. only happens when/if `xamarin_initialize / ObjCRuntime.Initialize` throw an native exception.
```
Comment (VincentDondain) [inline tools/mtouch/simlauncher.mm]:
```suggestion
	// 1. only happens when/if `xamarin_initialize / ObjCRuntime.Initialize` throws a native exception.
```

File: tools/mtouch/simlauncher.mm [hunk 3]
```diff
@@ -40,6 +40,14 @@ void xamarin_setup_impl ()
...
+	//    This should not happen (nor be ignored) since the runtime would not be fully initialized;
+	// 2. only happens when (non default) `MarshalManagedExceptionModeThrowObjectiveCException` is used,
+	//    specifically asking for the native exception to be thrown; and
+	// 3. only happens on simulator builds (hint: filename), i.e. not on code that ships to customers;
+	// 4. catching the exception creates an empty/white app, which hides/confuse the issue for developers.
```
Comment (VincentDondain) [inline tools/mtouch/simlauncher.mm]:
```suggestion
	// 4. catching the exception creates an empty/white app, which hides the issue or confuses developers.
```

File: tools/mtouch/simlauncher.mm [hunk 4]
```diff
@@ -48,4 +56,4 @@ void xamarin_setup_impl ()
 void xamarin_initialize_callbacks ()
 {
 	xamarin_setup = xamarin_setup_impl;
-}
\ No newline at end of file
+}
```
Comment (VincentDondain) [inline tools/mtouch/simlauncher.mm]: Incorrect line ending update (;

**Pull request merged**